### PR TITLE
DEP: pysatNASA instruments

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Partially addresses #697 by adding deprecation warnings to all Instrument modules and methods associated with the pysatNASA package.

## Type of change

- This change requires a documentation update

# How Has This Been Tested?

Added unit tests for all currently tested routines.
# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ n/a] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Review by aburrell — CHANGES_REQUESTED** [review]
The routines in the instrument sub-modules that are not fed into the Instrument class need deprecation warnings (e.g., omni_hro) and then any that have unit tests need the warnings tested (not bothering to add deprecation tests to those who aren't already covered).  

Additionally, in my package pulls I deprecated the relevant pysat.instrument.methods subpackages.  Are you planning on doing that here or in a separate pull request?

**Comment by jklenzing** [comment]
@aburrell, the only thing left is the deprecation warnings on the constellations.  I can't run the code locally to verify, travis has it still failing.  Any thoughts?

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,9 +23,14 @@ This project adheres to [Semantic Versioning](http://semver.org/).
       - the pysatMadrigal: jro_isr, dmsp_ivm
       - the pysatSpaceWeather: sw_dst, sw_f107, sw_kp
       - the pysatModels: ucar_tiegcm
+      - the pysatNASA: cnofs_ivm, cnofs_plp, cnofs_vefi, de2_lang, de2_nacs,
+        de2_rpa, de2_wats, icon_euv, icon_fuv, icon_ivm, icon_might, iss_fpmu,
+        omni_hro, rocsat1_ivm, sport_ivm, timed_saber, timed_see
    - The usage of a numeric string for `sat_id` to specify number of points
      in test instruments
-   - SpaceWeather, Incubator (DEMETER), and Madrigal instrument method
+   - SpaceWeather, Incubator (DEMETER), Madrigal, cdaweb, and icon instrument
```
Comment (aburrell) [inline CHANGELOG.md:31]:
```suggestion
   - SpaceWeather, Incubator (DEMETER), Madrigal, CDAWeb, and ICON instrument
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,9 +23,14 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+     methods
+   - The de2 and icon constellation objects
```
Comment (aburrell) [inline CHANGELOG.md:33]:
```suggestion
   - The DE2 and ICON Constellation objects
```

**Review by aburrell — COMMENTED** [review]
That's because the warnings are in the module, which isn't run.  There's no way to get them to be raised.  The Instrument warnings were raised after the init method was run.  But there's no equivalent for the Constellation 😞

File: pysat/constellations/de2.py [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+
+import pysat
+
+
+warnings.warn(" ".join(["This constellation",
+                        "has been removed from the pysat-managed",
+                        "Instruments in pysat 3.0.0, and now resides in",
+                        "pysatNASA:",
+                        "https://github.com/pysat/pysatNASA"]),
+              DeprecationWarning, stacklevel=2)
```
Comment (aburrell) [inline pysat/constellations/de2.py:20]:
```suggestion 
import pysat
```

**Comment by jklenzing** [comment]
> That's because the warnings are in the module, which isn't run. There's no way to get them to be raised. The Instrument warnings were raised after the init method was run. But there's no equivalent for the Constellation 😞

Should we just ignore the constellation module?  If someone does try to run it, they should get the deprecation warnings from the instruments themselves.

File: pysat/constellations/de2.py (hunk 3 above)
Comment (aburrell) [reply pysat/constellations/de2.py:20]: Here and ICON

**Review by aburrell — COMMENTED** [review]
I think you can solve the problem by adding a warning to the constellation object that triggers if the DE2 or ICON constellations are used as input.  Similar to what you did for the unit test.

File: pysat/constellations/icon.py [hunk 4]
```diff
@@ -1,8 +1,23 @@
...
+
+import pysat
+
+
+warnings.warn(" ".join(["This constellation",
+                        "has been removed from the pysat-managed",
+                        "Instruments in pysat 3.0.0, and now resides in",
+                        "pysatNASA:",
+                        "https://github.com/pysat/pysatNASA"]),
+              DeprecationWarning, stacklevel=2)
```
Comment (jklenzing) [inline pysat/constellations/icon.py:20]:
```suggestion
```

Comment (aburrell) [reply pysat/constellations/icon.py:20]: For future reference, you can edit my suggestion.

**Comment by jklenzing** [comment]
🤞

**Comment by jklenzing** [comment]
Update: The ICON constellation object does not work, and has not for a while.  I've documented this in pysat/pysatNASA#49 to decide on future standards.  Looking through old versions, this would have been broken in python 2.2.0.

**Comment by aburrell** [comment]
> Update: The ICON constellation object does not work, and has not for a while. I've documented this in pysat/pysatNASA#49 to decide on future standards. Looking through old versions, this would have been broken in python 2.2.0.

Can it be minimally fixed here?

**Comment by jklenzing** [comment]
I've pushed a simple version through here.  If that works, I'll try to put together a fixed version.

**Comment by jklenzing** [comment]
Update: this is not as bad as I thought.  Pushing through a version with 8 instruments.  The other 4 are the component winds that go into the vector winds products, so i think we can leave these off.  In a practical sense, I suspect anyone wanting to run the constellation will probably just build a custom one with the bits they want to compare.

**Comment by jklenzing** [comment]
Another difference that may need to be addressed here: pysatNASA uses the CDAWeb archive for ICON, pysat 2.x uses the Berkeley FTP server.  This results in a slight difference in file names (mixed case, underscores), which means that the data path migration system will not work for users who migrate from 2 to 3.  We should probably include this in the deprecation warning here. 

The alternative is to do the server switch here, which would basically introduce a breaking change in 2.3.

**Comment by aburrell** [comment]
Just add that info to the deprecation warning.  It'd be good to add a "migration guide" page to the pysatNASA docs.

**Comment by jklenzing** [comment]
proposed addition to warnings: 
"Note that the ICON files will be retrieved from the CDAWeb server in pysatNASA, please see the migration guide there for more details."

We can add a link once I get the docs up there, or just leave it like this.

**Comment by aburrell** [comment]
Counterproposal: "Note that the ICON files are retrieved from the Berkley FTP server here and from the CDAWeb server in pysatNASA, resulting in a difference in local file names.  Please see the migration guide there for more details."

**Comment by jklenzing** [comment]
"Coverage decreased (-6.0e-05%)" 😅

**Review by aburrell — APPROVED** [review]
Nice revisions, they all look good to me!

**Pull request merged**